### PR TITLE
Added Weibull and half-Cauchy distributions and tests

### PR DESCRIPTION
**Pull request by fonnesbeck**

### CODE REVIEW

File: pymc/distributions/continuous.py [hunk 1]
```diff
@@ -363,6 +392,44 @@ def logp(self, value):
...
+      - :math:`E(x)=\beta \Gamma(1+\frac{1}{\alpha})`
+      - :math:`median(x)=\Gamma(\log(2))^{1/\alpha}`
+      - :math:`Var(x)=\beta^2 \Gamma(1+\frac{2}{\alpha} - \mu^2)`
+
+    """
+    def __init__(self, alpha, beta, *args, **kwargs):
+        super(Weibull, self).__init__(*args, **kwargs)
+        self.alpha = alpha
+        self.beta = beta
+        self.mean = beta * exp(gammaln(1 + 1./alpha))
```
Comment (jsalvatier) [inline pymc/distributions/continuous.py:418]: Looks like we should get an actual gamma function? I think gammaln is relatively slow.

File: pymc/tests/test_distributions.py [hunk 2]
```diff
@@ -197,11 +197,32 @@ def test_cauchy():
...
 def test_gamma():
     pymc_matches_scipy(
             Gamma, Rplus, {'alpha': Rplusbig, 'beta': Rplusbig},
-            lambda value, alpha, beta: sp.gamma.logpdf(value, alpha, scale = 1.0/beta)
+            lambda value, alpha, beta: sp.gamma.logpdf(value, alpha, scale=1.0/beta)
+            )
+
+def scipy_exponweib_sucks(value, alpha, beta):
+    # This function is required because SciPy's implementation of
+    # the Weibull PDF sucks
```
Comment (jsalvatier) [inline pymc/tests/test_distributions.py]: haha, great function name, but your comment should say how it sucks.

**Comment by jsalvatier** [comment]
This looks good to me. I assume those Potential changes are the same ones already committed now?

File: pymc/distributions/continuous.py (hunk 1 above)
Comment (fonnesbeck) [inline pymc/distributions/continuous.py:418]: This is the only place it gets used at present, but yes we should.

File: pymc/tests/test_distributions.py (hunk 2 above)
Comment (fonnesbeck) [inline pymc/tests/test_distributions.py]: Done. Have also created an issue on the scipy repo.

Comment (jsalvatier) [inline pymc/tests/test_distributions.py]: Great! I would include that in the commit message.

**Comment by fonnesbeck** [comment]
The problem with exponweib in Scipy has now been fixed in master, but I am going to leave things as-is until the fix makes its way to a release.

**Pull request merged**